docs/config/lua/wezterm.table/merge.md

```diff
@@ -0,0 +1,36 @@
+# `wezterm.table.merge(array_of_tables [, keep_first])`
+
+{{since('nightly')}}
+
+This function merges Lua tables passed to it in the form of an array.
+I.e., to merge the Lua tables `tbl1` and `tbl2`, we can pass them to
+the function as `{ tbl1, tbl2 }`. (See below.)
+
+By default this function merges tables with identical keys by taking
+the value from the last table in the array with each given key.
+
+The optional `keep_first` allows us to instead prefer values from the
```

Comment:
boolean arguments can be hard to intuit the effect, especially if the call looks something like `merge({a, b, c, d, e}, true)`.

Can you remove the boolean argument and split this into two separate lua functions: `merge` and `merge_first`?  (I'm also open to better naming of `merge_first`! Are there examples in other languages/runtimes that might inform naming for this?)
I think that will lead to much clearer code.

Reply:
Neovim has `vim.tbl_extend` & `vim.tbl_deep_extend`, where you pass the behavior you want in first param
https://neovim.io/doc/user/lua.html#vim.tbl_extend()
https://neovim.io/doc/user/lua.html#vim.tbl_deep_extend()

Reply:
> Can you remove the boolean argument and split this into two separate lua functions: `merge` and `merge_first`? (I'm also open to better naming of `merge_first`! Are there examples in other languages/runtimes that might inform naming for this?) I think that will lead to much clearer code.

I can split it into two, but will wait until we agree on the details. Would you prefer two functions over string arguments like neovim?

For naming if we go with two functions, maybe `merge_keep` (more like the neovim naming) is better than `merge_first`? But this is not something I feel strongly about.

Reply:
Hmm, ok, you could add:

```rust
#[derive(Debug, FromDynamic, ToDynamic, Clone, PartialEq, Eq)]
enum ConflictMode {
   Keep,
   Force,
   Error,
}
impl_lua_conversion_dynamic!(ConflictMode);
```

then make the parameter an `Option<ConfictMode>`.  That should allow passing `"Keep"` as that parameter value. If that can be the first parameter without being unusable (eg: not requiring that the user do `merge(nil, {a, b})` that would be fine, but otherwise it could be the last parameter.

If we want to be influenced by nvim here, we could also rename this to `extend`.

Reply:
> Hmm, ok, you could add:
> 
> ```rust
> #[derive(Debug, FromDynamic, ToDynamic, Clone, PartialEq, Eq)]
> enum ConflictMode {
>    Keep,
>    Force,
>    Error,
> }
> impl_lua_conversion_dynamic!(ConflictMode);
> ```

This code gives errors. Also, from what I can see isn't that macro for structs and not enums? (I am not used to reading rust macros)

I can just implement what is needed myself for this one case though.